ui/statistics/src/main/java/de/danoeh/antennapod/ui/statistics/years/BarChartView.java

```diff
@@ -89,17 +89,19 @@ public void draw(@NonNull Canvas canvas) {
             paintBars.setStrokeWidth(height * 0.015f);
             paintBars.setColor(colors[0]);
             int colorIndex = 0;
-            int lastYear = data.size() > 0 ? data.get(0).getYear() : 0;
+            int lastYear = data.size() > 0 ? data.get(data.size() - 1).getYear() : 0;
+            int prevYear = data.size() > 0 ? data.get(0).getYear() : 0;
+            int timeSpan = lastYear - prevYear;
+            boolean firstYearDisplay = timeSpan < 3;
+
             for (int i = 0; i < data.size(); i++) {
                 float x = textPadding + (i + 1) * stepSize;
-                if (lastYear != data.get(i).getYear()) {
-                    lastYear = data.get(i).getYear();
+                if ((prevYear != data.get(i).getYear()) || ((i == 0) && firstYearDisplay)) {
```

Comment:
What if the data starts in December? Doesn't showing the date then clash with the next year? Maybe the first year should only be displayed if the first month is earlier than, say, October (one would need to test with 3 years and October to see if the texts clash